core/src/main/java/org/apache/iceberg/BaseRewriteFiles.java

```diff
@@ -57,4 +57,22 @@ public RewriteFiles rewriteFiles(Set<DataFile> filesToDelete, Set<DataFile> file
 
     return this;
   }
+
+  @Override
+  public RewriteFiles rewriteDeletes(Set<DeleteFile> deletesToDelete, Set<DeleteFile> deletesToAdd) {
+    Preconditions.checkArgument(deletesToDelete != null && !deletesToDelete.isEmpty(),
+        "Files to delete cannot be null or empty");
+    Preconditions.checkArgument(deletesToAdd != null && !deletesToAdd.isEmpty(),
```

Comment:
This check is incorrect, because if all the equality deletes are not hit the data files, then there will be no position delete to produce..

Reply:
I will suggest to add an unit test for this.

Reply:
I understand your concern. The check is used to discard the invalid rewrite, we don't want to continue the rewrite if there is no position delete produced. Don't we?

Reply:
This kind of rewrite is valid actually because it replace all the useless equality files to empty position delete files. After the rewrite action,   the normal read path don't have to filter the useless equality deletes again, that will be a great performance improvement.  So we have to submit the RewriteFiles transaction here.

Reply:
You could see the validation in the extended RewriteFiles API here ( https://github.com/apache/iceberg/pull/2294/files#diff-b92a78b7fb207d4979d503a442189d9d096e4d19519a4b83eed9e1e779843810R68)

Reply:
Make sense to me! I will update then.